x-pack/plugins/security_solution/server/usage/detections_helpers.ts

```diff
@@ -0,0 +1,148 @@
+      };
+    } else if (!isEnabled && isElastic) {
+      return {
+        ...stats,
+        detection_rules_elastic_disabled: stats.detection_rules_elastic_disabled + 1,
+      };
+    } else if (isEnabled && !isElastic) {
+      return {
+        ...stats,
+        detection_rules_custom_enabled: stats.detection_rules_custom_enabled + 1,
+      };
+    } else if (!isEnabled && !isElastic) {
+      return {
+        ...stats,
+        detection_rules_custom_disabled: stats.detection_rules_custom_disabled + 1,
+      };
+    } else {
+      return stats;
+    }
+  }, initialRuleUsage);
+
+export const buildMlJobUsage = (jobMetrics: DetectionsMetric[]): MlJobsUsage =>
+  jobMetrics.reduce((stats, { isEnabled, isElastic }) => {
+    if (isEnabled && isElastic) {
+      return {
+        ...stats,
+        ml_jobs_elastic_enabled: stats.ml_jobs_elastic_enabled + 1,
+      };
+    } else if (!isEnabled && isElastic) {
+      return {
+        ...stats,
+        ml_jobs_elastic_disabled: stats.ml_jobs_elastic_disabled + 1,
+      };
+    } else if (isEnabled && !isElastic) {
+      return {
+        ...stats,
+        ml_jobs_custom_enabled: stats.ml_jobs_custom_enabled + 1,
+      };
+    } else if (!isEnabled && !isElastic) {
+      return {
+        ...stats,
+        ml_jobs_custom_disabled: stats.ml_jobs_custom_disabled + 1,
+      };
+    } else {
+      return stats;
+    }
+  }, initialMlJobUsage);
+
+export const fetchRules = async (
+  index: string,
+  callCluster: LegacyAPICaller
+): Promise<DetectionsMetric[]> => {
+  let ruleMetrics: DetectionsMetric[] = [];
+  const ruleSearchOptions: SearchParams = {
+    body: { query: { bool: { filter: { term: { 'alert.alertTypeId': SIGNALS_ID } } } } },
+    filterPath: ['hits.hits._source.alert.enabled', 'hits.hits._source.alert.tags'],
+    ignoreUnavailable: true,
+    index,
+    size: 10000, // elasticsearch index.max_result_window default value
+  };
+  try {
+    const ruleResults = await callCluster<{ alert: { enabled: boolean; tags: string[] } }>(
+      'search',
+      ruleSearchOptions
+    );
+
+    if (ruleResults.hits?.hits?.length > 0) {
+      ruleMetrics = ruleResults.hits.hits.map((hit) => ({
+        isElastic: isElasticRule(hit._source.alert.tags),
+        isEnabled: hit._source.alert.enabled,
+      }));
+    }
+  } catch (e) {
+    // ignore failure, rules will be empty
+  }
+
+  return ruleMetrics;
+};
+
+export const fetchJobs = async (ml: MlPluginSetup | undefined): Promise<DetectionsMetric[]> => {
+  let jobMetrics: DetectionsMetric[] = [];
+
+  if (ml) {
+    try {
+      const mlCaller = ml.mlClient.callAsInternalUser;
+      const modules = await new DataRecognizer(
+        mlCaller,
+        ({} as unknown) as SavedObjectsClient
+      ).listModules();
+      const moduleJobs = modules.flatMap((module) => module.jobs);
+      const jobs = await jobServiceProvider(mlCaller).jobsSummary(['siem']);
+
+      jobMetrics = jobs.map((job) => ({
```

Comment:
Same thing here, but not sure how much of an issue it'll be to loop over all this data twice. I know we have about 60s till the usageCollector times out, but I don't _think_ we'll run into that